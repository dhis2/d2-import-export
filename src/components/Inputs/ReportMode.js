import i18n from '@dhis2/d2-i18n'
import React from 'react'
import { RadioGroup } from '../FinalFormComponents/RadioGroup'

export const OPTION_ERRORS = { value: 'ERRORS', label: i18n.t('ERRORS') }
export const OPTION_FULL = { value: 'FULL', label: i18n.t('Full') }
export const OPTION_DEBUG = { value: 'DEBUG', label: i18n.t('DEBUG') }
export const REPORT_MODE_KEY = 'reportMode'
export const REPORT_MODE_DEFAULT_VALUE = OPTION_ERRORS.value

export const ReportMode = () => (
    <RadioGroup
        name={REPORT_MODE_KEY}
        label={i18n.t('Report mode')}
        options={[OPTION_ERRORS, OPTION_FULL, OPTION_DEBUG]}
    />
)
