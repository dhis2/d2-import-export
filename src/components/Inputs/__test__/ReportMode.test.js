import React from 'react'
import toJson from 'enzyme-to-json'
import { shallow } from 'enzyme'
import { ReportMode } from '../ReportMode'

jest.mock('react-final-form')

describe('Input component - ReportMode', () => {
    it('should render correctly', () => {
        const file = shallow(<ReportMode />)

        expect(toJson(file)).toMatchSnapshot()
    })
})
