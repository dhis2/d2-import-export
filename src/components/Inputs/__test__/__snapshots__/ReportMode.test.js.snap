// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input component - ReportMode should render correctly 1`] = `
<RadioGroup
  dataTest="input-report-mode"
  label="Report mode"
  name="reportMode"
  options={
    Array [
      Object {
        "label": "Error",
        "value": "ERRORS",
      },
      Object {
        "label": "Full",
        "value": "FULL",
      },
      Object {
        "label": "Debug",
        "value": "DEBUG",
      },
    ]
  }
/>
`;
